packages/flex-plugin-e2e-tests/src/utils/browser.ts

```diff
@@ -1,314 +1,75 @@
-      locator = this.pageObjects.flex.adminDashboard.welcomeBanner;
-    } else if (view === 'Agent Desktop') {
-      locator = this.pageObjects.flex.agentDesktop.noTaskCanvas;
-    } else {
-      throw new Error(`${view} is not a valid view`);
-    }
-
-    await this.elementExistsAndIsVisible(locator, view, DEFAULT_PAGE_LOAD_TIMEOUT, DEFAULT_PAGE_LOAD_TIMEOUT);
-  }
-
-  /**
-   * Checks that element exists in DOM and is visible in UI
-   * @param locator selector wrapped in By
-   * @param elementName name of the searchable element
-   * @param existTimeout time to wait for until element is located in DOM
-   * @param visibleTimeout time to wait for until element is visible in the UI
-   */
-  private static async elementExistsAndIsVisible(
-    locator: By,
-    elementName: string,
-    existTimeout = DEFAULT_LOCATE_TIMEOUT,
-    visibleTimeout = DEFAULT_LOCATE_TIMEOUT,
-  ): Promise<void> {
-    const element = await this.elementExists(locator, elementName, existTimeout);
-    await this.elementIsVisible(element, elementName, visibleTimeout);
-  }
-
-  /**
-   * Check that element exists in the DOM
-   * @param locator selector wrapped in By
-   * @param elementName name of the searchable element
-   * @param timeout time to wait for until element is located in DOM
-   */
-  private static async elementExists(
-    locator: By,
-    elementName: string,
-    timeout = DEFAULT_LOCATE_TIMEOUT,
-  ): Promise<WebElement> {
-    try {
-      return await this.browser.wait(until.elementLocated(locator), timeout, `Could not find ${elementName} in DOM`);
-    } catch (e) {
-      logger.error(`Did not find locator [${locator}] for element: ${elementName} in DOM`);
-      throw new Error(JSON.stringify(e));
-    }
+  private static _attachNetworkInterceptor(): void {
+    this._page.on('request', async (request): Promise<void> => {
+      // Ignoring any 3rd part requests to speed up Twilio console load time
+      const url = request.url();
+      const re = new RegExp(this._domainsToInclude.join('|'));
+
+      if (request.resourceType() === 'script' && !re.test(url)) {
+        return request.abort();
+      }
+      return request.continue();
+    });
   }
 
   /**
-   * Check that element is visible
-   * @param element element which should be visible
-   * @param elementName name of the searchable element
-   * @param timeout time to wait for until element is visible in the UI
+   * Attach network interceptor
    */
-  private static async elementIsVisible(
-    element: WebElement,
-    elementName: string,
-    timeout = DEFAULT_LOCATE_TIMEOUT,
-  ): Promise<void> {
-    try {
-      await this.browser.wait(until.elementIsVisible(element), timeout, `Could not find ${elementName} in DOM`);
-    } catch (e) {
-      logger.error(`Element: ${elementName} is not visible in the UI`);
-      throw new Error(JSON.stringify(e));
-    }
+  private static _attachLogListener(): void {
+    this._page.on('console', (msg: ConsoleMessage) => {
```

Comment:
Do we not to remove these listeners at some point? You are adding listeners without ever removing them

Reply:
Ah, good catch. I even left a comment to address that but must have gotten distracted by irl. 😓